[WIP] Documentation improvements

atlz253 opened the pull request:
Documentation improvements made during the Yandex interface development summer school. At the moment, various errors may occur during use.

## Changelog

- [X]  added [doxygen awesome](https://github.com/jothepro/doxygen-awesome-css)
- [X]  added README for frontend developers
- [X]  next/prev page button style improvements
- [X]  new incredible documentation design
- [X]  search improvement
- [X]  table of contents improvements
- [X]   adaptation to different screens
- [ ]   new landing page [WIP]
- [X]  bug fixes and various code improvements

## Authors

- 👨🏻‍💻 frontend developers: @atlz253 @ZenkoWu @fleshr @InfinityScripter 
- 🧑🏼‍🎨 designers: @hellenisaeva @MariaGrinchenko
- 👨🏻‍💼 manager: @0GE1
- 👩🏻‍💼 marketing specialist: @makachusha

fleshr committed:
feat docs: change nav buttons style

fleshr committed:
feat docs: doxygen awesome theme

ZenkoWu committed:
feat docs: plug in custom styles

ZenkoWu committed:
fix docs: fix light dark-mode for gh-logo

InfinityScripter committed:
feat docs: update landing page styles

atlz253 committed:
docs: add README for frontenders

InfinityScripter committed:
feat docs: add border-radius hover btn

atlz253 committed:
refactor docs: moved the features to separate script files

InfinityScripter committed:
feat docs: update landing page styles

ZenkoWu committed:
fix docs: fix navigation button scripts

return scripts to footer.html and add extra file to doxygen.conf

ZenkoWu committed:
feat docs: remove search modal window

atlz253 committed:
feat docs: all results button

fleshr committed:
fix docs: toc did not appear on all pages

InfinityScripter committed:
feat docs: documentation page styles

atlz253 committed:
feat docs: change table of contents style

ZenkoWu committed:
feat docs: update header style

replace icons, change header color, change color text

atlz253 committed:
fix docs: color scheme

ZenkoWu committed:
fix docs: fix scripts for adding tg and gh header links

ZenkoWu committed:
style docs: format header opensource file

ZenkoWu committed:
feat docs: update header style

atlz253 committed:
feat docs: change font to Roboto

InfinityScripter committed:
fix docs: fix arrow color in tab menu header

InfinityScripter committed:
fix docs: fix TOC on all screen resolutions

fleshr committed:
feat docs: highlighting search results

InfinityScripter committed:
fix docs: searchbox position

ZenkoWu committed:
fix docs: tg and gh header links

ZenkoWu committed:
feat docs: remove header links hover

InfinityScripter committed:
feat docs: change modal style

atlz253 committed:
fix docs: landing text color

ZenkoWu commented:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/

fleshr commented:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/

InfinityScripter commented:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/

atlz253 commented on scripts/docs/scripts/toc.js:
```diff
@@ -0,0 +1,110 @@
+      $(".header"),
+      $("<div>")
+        .addClass("contents")
+        .append([toc, $(".contents").attr("class", "textblock")]),
```
We try to fix the warning in the future

apolukhin commented on scripts/docs/en/landing.md:
```diff
@@ -23,14 +17,14 @@ The problem of efficient I/O interactions is solved transparently for the
 
 <div class="landing-text">
-<pre style="padding: 20px; color: silver;white-space: pre-wrap; word-wrap: break-word;">
+<div style="font-family: monospace;">
```
Из-за этой правки теряются переносы строк ииедет форматирование кода. Но вы и так об этом уже знаете :)

atlz253 replied:
Исправим этот момент при переработке лендинга)

apolukhin replied:
тогда лучше пока ревертнуть правку, чтобы в бете документации отображалось по привычному и пользователи не закидали однотипными багрепортами, что "форматирование на лендинге поползло"

atlz253 replied:
В ближайшее время исправим

atlz253 replied:
@fleshr начал исправление

atlz253 replied:
Исправили

apolukhin commented on scripts/docs/scripts/codeHighlight.js:
```diff
@@ -0,0 +1,56 @@
+function highlight_code() {
```
Несколько небольших js файликов кажется что будут грузиться дольше чем один побольше. Можно в Makefile добавить команды для минимизации JS и склеивания маленьких файликов в один. Но это не обязательно :)

atlz253 replied:
Завёл под это отдельную задачку в трекере, думаю стоит в целом посмотреть как можно оптимизировать полученные файлы. Как идея: можно создать отдельные файлы header_release.html и footer_release.html, где будут ссылки на бандл скриптов и стилей.

apolukhin commented on scripts/docs/doxygen-awesome-css/doxygen-awesome-fragment-copy-button.js:
```diff
@@ -0,0 +1,85 @@
+/**
+
```
Надо в директорию scripts/docs/doxygen-awesome-css/ положить файлик Readme.md и указть в нем репозиторий, откуда брали исходники, а главное - версию/релиз исходников и описать, если накладывали патчи

Ещё надо в ./THIRD_PARTY.md добавить информацию в нужный раздел

atlz253 replied:
Готово)

apolukhin commented on THIRD_PARTY.md:
```diff
@@ -37,6 +37,9 @@ The above copyright notice and this permission notice shall be included in all c
 
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WI ...
 
+**[Doxygen Awesome](https://github.com/jothepro/doxygen-awesome-css)**
```
Нужно перенести повыше, под 
```
**[CPM library](https://github.com/cpm-cmake/CPM.cmake)**
Copyright (c) 2019-2022 Lars Melchior and contributors
```

atlz253 replied:
Готово)

apolukhin commented on scripts/docs/customdoxygen.css:
```diff
@@ -158,45 +46,238 @@ a.titlelink {
+        padding: 10px;
+    }
+
+    /* body, table, div, p, dl {
```
Неиспользуемые куски кода лучше потереть... Иначе они тут останутся на долго, и будут мозолить глаза

atlz253 replied:
Вообще нужно причесать этот файл

atlz253 replied:
Причесал customdoxygen.css, чтобы в нем можно было разбираться

apolukhin commented on scripts/docs/customdoxygen.css:
```diff
@@ -242,55 +339,251 @@ div.fragment {
-    opacity: 0.25;
-    font-size: 80%; /* default in insanely huge */
+html {
+    --primary-color: #f18128;
```
Возможно стоит подкрутить все `f18128 ` в более тёмную сторону, чтобы избавиться от белёсости цвета на некоторых мониторах

atlz253 replied:
Дизайнерам выдана соответствующая задача ;)

apolukhin commented on scripts/docs/customdoxygen.css:
```diff
@@ -158,45 +46,238 @@ a.titlelink {
+}
+
+#MSearchBox {
+    display: none;
```
Поиск по сайту очень важен, его всегда надо показывать на всех устройствах.

P.S.: лайк за большую поисковую строку на десктопе!

atlz253 replied:
@ZenkoWu Уже занимается

atlz253 replied:
Сделано)

atlz253 committed:
docs: doxygen awesome README

fleshr committed:
fix docs: landing codeblock layout

atlz253 committed:
feat docs: update highlight.js v11.8.0

ZenkoWu committed:
feat docs: header adaptive design

atlz253 committed:
refactor docs: header init script

atlz253 committed:
fix docs: broken styles

atlz253 committed:
style docs: format customdoxygen.css

ZenkoWu committed:
fix docs: fix tg and gh links position

atlz253 committed:
fix docs: fix landing page header visibility

InfinityScripter committed:
feat docs: landing page adaptive layout

change styles for adaptive design, fix gh-logo white/black theme, fix landing page style for landing-container

ZenkoWu committed:
fix docs: change border-radius for navigation buttons

fleshr committed:
feat docs: sidenav documentaion tree

atlz253 committed:
fix docs: fix burger menu work

ZenkoWu committed:
fix  docs: change iframe background color

atlz253 committed:
feat docs: new header layout

ZenkoWu committed:
fix docs: change line-height for toc

InfinityScripter committed:
feat docs: update code examples style

change style block code , fix margin for adaptive design, change color hljs-comment

atlz253 committed:
fix docs: fix header mobile and pad layout

atlz253 committed:
fix docs: fix desktop layout

atlz253 committed:
fix docs: fix dark mode toggle move

ZenkoWu committed:
fix docs: change reference modal colors

InfinityScripter committed:
fix docs: change code indentation

InfinityScripter committed:
fix docs: gh-logo black/white theme

InfinityScripter committed:
fix docs: code-hljs change color

ZenkoWu committed:
fix docs: remove bottom hr line

InfinityScripter committed:
fix docs: search-highlight-text

apolukhin commented on scripts/docs/customdoxygen.css:
```diff
@@ -1,138 +1,662 @@
-      top: 0;
-      width: 100%;
-    }
+html.dark-mode div.contents .dotgraph iframe {
```
Добавьте комент, что оно делает. И почему нужны iframe?

atlz253 replied:
@ZenkoWu прокомментирует. Коммент потерся в жариких ребейзах

atlz253 replied:
Готово)

apolukhin commented on scripts/docs/customdoxygen.css:
```diff
@@ -1,138 +1,662 @@
-      position: fixed !important;
-      top: 30px !important;
-    }
+/* Highlight.js overrides */
```
Для темы highlight.js есть отдельный файлик. Почему не подходит его использование?

atlz253 replied:
Мы решили перезаписывать стили highlight.js в отдельном файле, чтобы не потерять возможность обновления библиотеки в будущем

apolukhin commented on scripts/docs/customdoxygen.css:
```diff
@@ -1,138 +1,662 @@
-    padding: 13px 13px 13px 20px;
-    margin: 13px 13px 13px 20px;
-    max-width: 300px;
+.gh-logo-landing {
```
Тут что-то сломано :(

atlz253 replied:
Ломается в safari, будем исправлять)

apolukhin commented on scripts/docs/highlight.js/highlightjs.css:
```diff
@@ -0,0 +1,86 @@
+/*!
```
Зачем этот файл, если мы все цвета переопределяем в кастомной теме?

atlz253 replied:
Он остался для возможности обновления библиотеки в будущем. Лучше убрать его?

apolukhin commented on scripts/docs/layout_opensource.xml:
```diff
@@ -3,7 +3,118 @@
   <navindex>
     <tab type="mainpage" visible="no" title="Landing"/>
-    <tab type="user" visible="yes" url="de/d6a/md_en_2index.html" title="Documentation" intro=""/>
+    <tab type="usergroup" title="Documentation" url="@ref md_en_2index">
```
Поддерживать этот файл в таком виде будет очень мучительно. В доьавок, эти данные теперь надо поддерэивать консистентно иж в 3х местах: здесь, на главной, на каждой странице поддерживать переходы вперёд/назад

atlz253 replied:
Мы обсудили с @alexiprof сегодня этот момент. Решили пока оставить вот так, но конечно это не дело :(

ZenkoWu committed:
style docs: add comment to html.dark-mode div.contents .dotgraph iframe

ZenkoWu committed:
style docs: remove console.log from styledBtn.js

fleshr committed:
fix docs: hide summary in content header

InfinityScripter committed:
fix docs: changing code block stroke styles for adaptive design 767px

ZenkoWu committed:
feat docs: update styles for burger modal modal

atlz253 committed:
fix docs: fix scripts loading order

InfinityScripter committed:
fix docs: Search Results Window add for adaptive

InfinityScripter committed:
fix docs: change style for close btn in mobile adaptive

InfinityScripter committed:
fix docs: change-font-size-search-result

ZenkoWu committed:
feat docs: change nav buttons styles

InfinityScripter committed:
fix docs: MSearchField change on all width

InfinityScripter committed:
fix docs: change-code-style-hljs

fleshr committed:
fix docs: burger button style
